_ml-commons-plugin/agents-tools/index.md

```diff
@@ -151,6 +152,53 @@ POST /_plugins/_ml/agents/_register
 }
 ```
 
+### Plan, Execute, and Reflect agents
+**Introduced 3.0**
+{: .label .label-purple }
+
+Plan, Execute and Reflect agent is an experimental feature.
+
+The Plan, Execute, and Reflect agent is designed for solving complex tasks that require iterative reasoning and step-by-step execution. It uses one LLM to create and update a plan (planner) and another LLM (or the same one, by default) to execute each individual step through a built-in conversational agent.
+
+The Plan, Execute, and Reflect agent works in three phases:
+
+- **Planning** – The planner LLM generates an initial step-by-step plan using the available tools.
+- **Execution** – Each step is executed sequentially using the conversational agent and the available tools.
+- **Re-evaluation** – After executing each step, the planner LLM re-evaluates the plan using intermediate results. Therefore, the plan can be adjusted dynamically to skip, add, or change steps based on new context.
+
+Similarly to a conversational agent, the Plan, Execute, and Reflect agent stores the interaction between the LLM and the agent in a memory index. In the following example, the agent uses a `conversation_index` to persist the execution history, including the user’s question, intermediate results, and final outputs.
+
+The agent automatically selects the most appropriate tool for each step based on the tool descriptions and current context.
+
+The agent currently supports re-evaluation only after each step. This allows it to dynamically adapt the plan based on intermediate results before proceeding to the next step.
+
+```json
+POST /_plugins/_ml/agents/_register
+{
+  "name": "My Plan Execute Reflect Agent",
+  "type": "plan_execute_and_reflect",
+  "description": "Agent for dynamic task planning and reasoning",
+  "llm": {
+    "model_id": "YOUR_LLM_MODEL_ID",
+    "parameters": {
+      "prompt": "${parameters.question}"
+    }
+  },
+  "memory": {
+    "type": "conversation_index"
+  },
+  "parameters": {
+    "_llm_interface": "YOUR_LLM_INTERFACE"
```

Comment:
should we link to the doc `_ml-commons-plugin/api/agent-apis/register-agent.md` to let user find details more easily ?